packages/eslint-config/plugin/no-self-package-import.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Derive the `@deephaven` package name from the given path.
+ * Note that this assumes that the package folder name uses the same naming
+ * convention as the package.json `name` field.
+ */
+function deriveDeephavenPackageNameFromPath(path) {
+  // In the off chance that there is a packages folder higher up in the path,
+  // find the deepest one.
+  const packageFolderNameI = path.lastIndexOf('packages');
+
+  if (packageFolderNameI === -1) {
+    return null;
+  }
+
+  // Find the path token immediately after the `packages` folder.
+  const [packageFolderName] = path
+    .substring(packageFolderNameI + 'packages'.length + 1)
+    .split('/');
+
+  if (packageFolderName === '') {
+    return null;
+  }
+
+  return `@deephaven/${packageFolderName}`;
```

Comment:
One thing I'm not crazy about is this rule is not generic; even when we include this eslint config from Enterprise it won't technically be correct. We should also be using the `path` package to resolve paths rather than manipulating the string directly, as there's probably some edge cases you're missing here (e.g. what if I had Enterprise using this eslint config and it was in a folder like `~/deephaven/packages/app/...`, then it would think the package name was `@deephaven/app`).
Getting the package name from the package.json should be doable, though I do also wonder about performance and how we can cache results (since this will be called on every import in every file, don't want to have to traverse and read package.json every time...). I'd rather check in the fix first without the eslint rule than check it in with specific `@deephaven` logic.

Reply:
This "shouldn't" ever actually be used in enterprise since it doesn't really apply. It is currently consumed via the top level Community package so doesn't get pulled in with the shared config. It really is scoped to @deephaven community packages. Does this alleviate any of your concern?